### PR TITLE
PROD-30973: Implement "user cancels enrollment to event" event for the EDA

### PR DESCRIPTION
**Pull request by nkoporec**
## Problem (for internal)
Currently we don't have user cancels enrollment to the event.

## Solution (for internal)
Add this event.

**The PR is based on this branch https://github.com/goalgorilla/open_social/pull/4136, so it needs to get merged first.**

## How to test
1.Set up Open Social with broker (kafka) and kafka-ui for viewing messages
2.Enable the social_eda_dispatcher module
3.Create a new event
4. As a user join the event
5. Check the Kafka UI for a new topic and message for new enrollment.
6. As the same user cancels the enrollment via the cancel enrollment form
7. Check the Kafka UI for a new cancel event and test that it matches the spec

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4142-yfk627lhadm8jzixir9wtmcaf41mthrr.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67215e2d2737e5f7e52d014a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4142-yfk627lhadm8jzixir9wtmcaf41mthrr.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67215e2d2737e5f7e52d014a

File: modules/social_features/social_event/tests/src/Unit/EdaEventEnrollmentHandlerTest.php [hunk 1]
```diff
@@ -306,6 +306,28 @@ public function testEventEnrollmentCreate(): void {
     $handler->eventEnrollmentCreate($this->eventEnrollment);
   }
 
+  /**
+   * Tests the eventEnrollmentCreate method.
+   *
+   * @covers ::eventEnrollmentCreate
+   */
+  public function testEventEnrollmentCancel(): void {
+    $eventEnrollment = $this->createMock(EventEnrollmentInterface::class);
```
Comment (tregismoreira) [inline modules/social_features/social_event/tests/src/Unit/EdaEventEnrollmentHandlerTest.php:315]: Do we need this?

**Review by tregismoreira — APPROVED** [review]

**Comment by ribel** [comment]
Hi @nkoporec,

1. Please add the ticket number to the prefix of commit and link in the issue tracker.
2. Please check the comment in the ticket.
3. And it would be helpful to have an example of generated payload as in https://github.com/goalgorilla/open_social/pull/4114

Thank you 🙇

**Pull request closed without merging**